### PR TITLE
wifi: esp32: enhance handling of AP connect events

### PR DESCRIPTION
**Pull request by ndrs-pst**
This commit aims to improve the integration of `esp_wifi_drv` by
providing the link mode information to `wifi_mgmt` when a station device
is connected to the AP.

### CODE REVIEW

**Comment by marekmatej** [comment]
hi @ndrs-pst thanks for the contribution! Would you mind re-write the commit message, so it is more clear what was intended to achieve here?

**Comment by ndrs-pst** [comment]
Yes, for sure. So if you can guide the direction of the message it would be great.

File: drivers/wifi/esp32/src/esp_wifi_drv.c [hunk 1]
```diff
@@ -729,19 +763,19 @@ static int esp32_wifi_dev_init(const struct device *dev)
...
+	.iface_status  = esp32_wifi_status,
 #if defined(CONFIG_NET_STATISTICS_WIFI)
-	.get_stats	   = esp32_wifi_stats,
+	.get_stats     = esp32_wifi_stats,
 #endif
 };
 
 static const struct net_wifi_mgmt_offload esp32_api = {
-	.wifi_iface.iface_api.init	  = esp32_wifi_init,
+	.wifi_iface.iface_api.init = esp32_wifi_init,
```
Comment (krish2718) [inline drivers/wifi/esp32/src/esp_wifi_drv.c:778]: This is just whitespace noise, please remove.

**Comment by marekmatej** [comment]
> Yes, for sure. So if you can guide the direction of the message it would be great.

Ok, so perhaps, instead of naming the function calls used (which is visible in the diff), you could write about actual improvement that change brings to the esp32 wifi management.

File: drivers/wifi/esp32/src/esp_wifi_drv.c (hunk 1 above)
Comment (ndrs-pst) [reply drivers/wifi/esp32/src/esp_wifi_drv.c]: Thanks for your feedback about the whitespace noise. I've already removed it from the commit.
By the way, could you let me know when or in what situations I can remove such whitespace?

Comment (krish2718) [reply drivers/wifi/esp32/src/esp_wifi_drv.c]: Well, if it was intentional, then you can move that to a separate commit with commit logs detailing the reason. The code review is done per-commit esp. for large PR, so, thumb rule is one logical change per-commit.

**Review by krish2718 — DISMISSED** [review]

**Review by marekmatej — DISMISSED** [review]

**Review by sylvioalves — DISMISSED** [review]

**Review by LucasTambor — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by ndrs-pst** [comment]
Sorry for the inconvenience. I just realized that the link mode assignment was wrong.
- phy_11n should be WIFI_4.
- phy_11g should be WIFI_3.

There is no 802.11a, which is WIFI_2. 🫡

File: drivers/wifi/esp32/src/esp_wifi_drv.c [hunk 2]
```diff
@@ -287,9 +287,37 @@ static void esp_wifi_handle_sta_disconnect_event(void *event_data)
 static void esp_wifi_handle_ap_connect_event(void *event_data)
 {
 	wifi_event_ap_staconnected_t *event = (wifi_event_ap_staconnected_t *) event_data;
+	wifi_sta_list_t sta_list;
+	struct wifi_ap_sta_info sta_info;
+
+	sta_info.link_mode = WIFI_LINK_MODE_UNKNOWN;
+	memcpy(sta_info.mac, event->mac, WIFI_MAC_ADDR_LEN);
+	sta_info.mac_length = WIFI_MAC_ADDR_LEN;
+	sta_info.twt_capable = false;           /* Only support in 802.11ax */
```
Comment (krish2718) [inline drivers/wifi/esp32/src/esp_wifi_drv.c:296]: another way would be to just `memset` the `sta_info` and only fill in non-default value.

Comment (ndrs-pst) [reply drivers/wifi/esp32/src/esp_wifi_drv.c:296]: Considering that most of the data is non-zero except for `twt_capable`, `memset` may not be the choice in this situation. 😅

**Review by marekmatej — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by uLipe — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by kartben** [comment]
Ping @sylvioalves

**Comment by kartben** [comment]
@sylvioalves Please take a look :)

**Comment by sylvioalves** [comment]
@ndrs-pst sorry it took me so long to check this, I just missed (a few times). LGTM. Would you mind taking the clang-format suggestions regarding comments and code style?

**Comment by ndrs-pst** [comment]
> @ndrs-pst sorry it took me so long to check this, I just missed (a few times). LGTM. Would you mind taking the clang-format suggestions regarding comments and code style?

Sure, I've already addressed it. :)

**Comment by ndrs-pst** [comment]
Ping @sylvioalves 👀

**Review by sylvioalves — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**